desktop/src/main/java/org/ergoplatform/desktop/wallet/addresses/WalletAddressesComponent.kt

```diff
@@ -0,0 +1,129 @@
+
+    private val addressesList = mutableStateOf(emptyList<WalletAddress>())
+    private val walletState = mutableStateOf<Wallet?>(null)
+    private val passwordAddAddresses = mutableStateOf(0)
+    private val detailsScreenState = mutableStateOf<WalletAddress?>(null)
+
+    private val uiLogic = object : WalletAddressesUiLogic() {
+        override val coroutineScope: CoroutineScope
+            get() = componentScope()
+
+        override fun notifyNewAddresses() {
+            addressesList.value = addresses
+            walletState.value = wallet
+        }
+
+        override fun notifyUiLocked(locked: Boolean) {
+            navHost.lockScreen.value = locked
+        }
+    }
+
+    private val detailsUiLogic = WalletAddressDialogUiLogic()
+
+    init {
+        val nodeConnector = WalletStateSyncManager.getInstance()
+        componentScope().launch {
+            // balance refresh for newly added addresses needs to be triggered by
+            // observing NodeConnector refresh. Since the singleAddressRefresh will
+            // always return its last saved value, initializing can be done here, too
+            nodeConnector.singleAddressRefresh.collect {
+                uiLogic.init(Application.database.walletDbProvider, walletConfig.id)
+            }
+        }
+    }
+
+    @Composable
+    override fun renderScreenContents(scaffoldState: ScaffoldState?) {
+        WalletAddressesScreen(
+            addressesList.value,
+            walletState.value,
+            if (uiLogic.canDeriveAddresses()) { num ->
+                addAddresses(num)
+            } else null,
+            onOpenDetails = { detailsScreenState.value = it }
+        )
+
+        val detailsAddress = detailsScreenState.value
+        if (detailsAddress != null) {
+            WalletAddressDetailsDialog(detailsAddress, scaffoldState,
+                onChangeName = { address, newName ->
+                    componentScope().launch {
+                        detailsUiLogic.saveWalletAddressLabel(
+                            Application.database.walletDbProvider,
+                            address.id,
+                            newName.ifEmpty { null }
+                        )
+                    }
+                    detailsScreenState.value = null
+                },
+                onDeleteAddress = {
+                    componentScope().launch {
+                        detailsUiLogic.deleteWalletAddress(Application.database, it.id)
+                    }
+                    detailsScreenState.value = null
+                },
+                onDismiss = { detailsScreenState.value = null }
+            )
+        }
+
+        if (passwordAddAddresses.value > 0) {
+            PasswordDialog(
+                onDismissRequest = { passwordAddAddresses.value = 0 },
+                onPasswordEntered = {
+                    proceedAuthFlowWithPassword(it, walletConfig, ::proceedFromAuthFlow)
+                }
```

Comment:
please comment in the code about lifecycle of the password.

Reply:
Done